### DISCUSSION TITLE
pipe shell command to UI

### DISCUSSION THREAD

**Discussion by og900aero** [gokcehan/lf#1865] [Q&A] [answered]
I use the following code, when I open archive file:

```
cmd open &{{
	case "$(echo "$f" | sed 's/\([A-Z]\)/\L\1/g')" in
	*.tar.bz|*.tar.bz2|*.tbz|*.tbz2|*.tar.gz|*.tgz|*.tar.zst|*.tar.xz|*.txz|*.zip|*.rar|*.iso)
			mntdir="$f-archivemount"
			if [ ! -d "$mntdir" ]; then
				mkdir "$mntdir"
				fuse-archive -o allow_other "$f" "$mntdir"
			fi
			mntdir="$(printf '%s' "$mntdir" | sed 's/\\/\\\\/g;s/"/\\"/g')"
			lf -remote "send $id cd \"$mntdir\""
			lf -remote "send $id reload"
	    	;;
```
I want to pipe fuse-archive' messages to UI (because show mount progress).
I try replace fuse-archive -o allow_other "$f" "$mntdir" the following code:
`lf -remote "send $id %fuse-archive -o allow_other \"$f\" \"$mntdir\""`
Works, but it have a problem with this solution:
It does not wait for the command to finish, but wants to enter the mounted directory immediately.
What is the right solution?

**Comment by CatsDeservePets** [comment, marked as answer]
I think this is not possible using `pipe-shell`. You have to use `shell-wait` (i.e. using the `!` prefix).

**Reply by og900aero** [reply]
Unfortunately, that's completely different because it shows it in a shell window. Anyway, then it can't be solved.
